src/components/Questionnaire/QuestionnaireEditor.tsx

```diff
@@ -699,9 +699,10 @@ export default function QuestionnaireEditor({ id }: QuestionnaireEditorProps) {
     mutationFn: mutate(questionnaireApi.update, {
       pathParams: { id: id! },
     }),
-    onSuccess: () => {
+    onSuccess: (data: QuestionnaireDetail) => {
       toast.success("Questionnaire updated successfully");
       queryClient.invalidateQueries({ queryKey: ["questionnaireDetail", id] });
+      navigate(`/admin/questionnaire/${data.slug}`);
```

Comment:
let's not add this for now. Questionnaire is slightly different anyways, since there's no draft feature, users might want to save and keep on editing.